include/SH3/system/glprogram.hpp

```diff
@@ -73,6 +72,87 @@ namespace sh3_gl
          */
         void Unbind();
 
+        /**
+         *  Get the location of a uniform in our compiled GL Program.
+         *
+         *  @param name - The name of the uniform we want the location of.
+         *
+         *  @return GLint - Location of this uniform, where -1
+         */
+         GLint GetUniformLoc(const std::string& name);
+
+        /**
+         *  Set the value of a 2 value matrix in this program
+         *
+         *  @param name - Name of the uniform we want to load the data into.
+         *  @param mat - The matrix we want to send to the program.
+         */
+         void SetUniformMat2(const std::string& name, const glm::mat2& mat);
```

Comment:
Yeah I'll get on that. I wasn't too sure if it would cause confusion, so I made it explicit which uniform type was being edited in the program.